src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/FrameworkEventSource.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Runtime.CompilerServices;
+
+namespace System.Diagnostics.Tracing
+{
+    // To use the framework provider
+    // 
+    //     \\clrmain\tools\Perfmonitor /nokernel /noclr /provider:8E9F5090-2D75-4d03-8A81-E5AFBF85DAF1 start
+    //     Run run your app
+    //     \\clrmain\tools\Perfmonitor stop
+    //     \\clrmain\tools\Perfmonitor print
+    //
+    // This will produce an XML file, where each event is pretty-printed with all its arguments nicely parsed.
+    //
+    [EventSource(Guid = "8E9F5090-2D75-4d03-8A81-E5AFBF85DAF1", Name = "System.Diagnostics.Eventing.FrameworkEventSource")]
+    internal sealed class FrameworkEventSource : EventSource
+    {
+        public static readonly FrameworkEventSource Log = new FrameworkEventSource();
+
+        // Keyword definitions.  These represent logical groups of events that can be turned on and off independently
+        // Often each task has a keyword, but where tasks are determined by subsystem, keywords are determined by
+        // usefulness to end users to filter.  Generally users don't mind extra events if they are not high volume
+        // so grouping low volume events together in a single keywords is OK (users can post-filter by task if desired)
+        public static class Keywords
+        {
+            public const EventKeywords ThreadPool = (EventKeywords)0x0002;
+            public const EventKeywords ThreadTransfer = (EventKeywords)0x0010;
+        }
+
+        /// <summary>ETW tasks that have start/stop events.</summary>
+        public static class Tasks // this name is important for EventSource
+        {
+            /// <summary>Send / Receive - begin transfer/end transfer</summary>
+            public const EventTask ThreadTransfer = (EventTask)3;
+        }
+
+        public static class Opcodes
+        {
+            public const EventOpcode ReceiveHandled = (EventOpcode)11;
+        }
+
+        // The FrameworkEventSource GUID is {8E9F5090-2D75-4d03-8A81-E5AFBF85DAF1}
+        private FrameworkEventSource() : base(new Guid(0x8e9f5090, 0x2d75, 0x4d03, 0x8a, 0x81, 0xe5, 0xaf, 0xbf, 0x85, 0xda, 0xf1), "System.Diagnostics.Eventing.FrameworkEventSource") { }
+
+        // optimized for common signatures (used by the ThreadTransferSend/Receive events)
+        [NonEvent]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Concurrency", "CA8001", Justification = "This does not need to be correct when racing with other threads")]
```

Comment:
Do we run this FxCop rule still?

Reply:
Presumably as there are many more occurrences of this attribute in shared

Reply:
It is left over from desktop. The tools that checks for it do not run anywhere anymore. It is fine to delete the `CA8001` from this file - it was deleted from majority of files that had it in desktop already.